### PR TITLE
Docs essentials

### PR DESCRIPTION
**Pull request by nour-borgi**
# Motivation

This PR includes the following: 
* Added tabbed code (see in screenshots)
* Added Essentials docs (Still recipes will be pushed in another PR) 

The tabbed code includes the following features: 
* Disable/enable showing the tabs by setting showTabs to true/false
* Include one or more tabs
* Languages supported are listed [here](https://prismjs.com/test.html#language=css&text=yarn%20add%20x%20)
* Copy the code with effects (showing "copied!" after copy)
* Different sizes: very-small (one line of code), small, middle, big. If you don't set a specific size, it will be dynamic (best case for dynamic when you have one tab).

image
image
image

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# Checklist:

- [ ] I have done the work for both react and vue
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (Storybook)
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/ez-react/src/docs/4_2_responsiveCharts.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import Tabs from './Tabs';
+
+<Meta
+  title="Get Started/Essentials/Responsive Charts"
+  parameters={{ options: { isToolshown: false }, id: 5 }}
+/>
+
+# Responsive Charts
+
+All chart recipes in **EazyChart** comes with fixed width,
```
Comment (deathsh0ot) [inline packages/ez-react/src/docs/4_2_responsiveCharts.stories.mdx:11]: All chart recipes in **EazyChart** come with fixed width,

**Review by deathsh0ot — CHANGES_REQUESTED** [review]
- background color of the embedded code should be applied to all the code container.
- Missing the package react-highlighter from package.json.

File: packages/ez-react/src/docs/4_3_commonProps.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    gdp: 4.872,
+    country: 'Japan',
+  },
+];`}
+  ></Tab>
+</Tabs>
+
+The column chart in this case is the best option.
+We should specify the domainKey of `xAxis` and `yAxis` to the corresponding name of the property.
+In our case, the `country` will be presented in the `xAxis` and `gdp` in the `yAxis`.
```
[the remark this answers is not in the document]
Comment (nour-borgi) [reply packages/ez-react/src/docs/4_3_commonProps.stories.mdx:50]: Done ! thanks!

File: packages/ez-react/src/docs/4_3_commonProps.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+easeCubic: 'Cubic' | easeCubicIn: 'Cubic In' | easeCubicOut: 'Cubic Out' | easeCubicInOut: 'Cubic In/Out' |
+easePoly: 'Polynomial' | easePolyIn: 'Polynomial In' | easePolyOut: 'Polynomial Out' | easePolyInOut: 'Polynomial In/Out' |
+easeSin: 'Sinusoidal' | easeSinIn: 'Sinusoidal In' | easeSinOut: 'Sinusoidal Out' | easeSinInOut: 'Sinusoidal In/Out' |
+easeExp: 'Exponential' | easeExpIn: 'Exponential In' | easeExpOut: 'Exponential Out' | easeExpInOut: 'Exponential In/Out' |
+easeCircle: 'Circle' | easeCircleIn: 'Circle In' | easeCircleOut: 'Circle Out' | easeCircleInOut: 'Circle In/Out' |
+easeBounce: 'Bounce' | easeBounceIn: 'Bounce In' | easeBounceOut: 'Bounce Out' | easeBounceInOut: 'Bounce In/Out' |
+easeBack: 'Back' | easeBackIn: 'Back In' | easeBackOut: 'Back Out' | easeBackInOut: 'Back In/Out' | easeElastic: 'Elastic' |
+easeElasticIn: 'Elastic In' | easeElasticOut: 'Elastic Out' | easeElasticInOut: 'Elastic In/Out'.
+
+This animations can be provided to the chart used as follows:
```
[the remark this answers is not in the document]
Comment (nour-borgi) [reply packages/ez-react/src/docs/4_3_commonProps.stories.mdx:109]: Done! thanks!

File: packages/ez-react/src/docs/4_3_commonProps.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+This component serves as a way to display information about the chart (section, point).
+You can add your custom tooltip component, or simply use the one provided by default.
+
+(Tooltip)[Link to tooltip image]
+
+## Legend
+
+The legend is an indicator of a portion of a chart,
+it is the information on this portion like the name, shape, and color of a point (shape) on a chart.
+The Legend component lets the user choose over what row you want to display in a bar chart, for example, or a pie chart.
```
[the remark this answers is not in the document]
Comment (nour-borgi) [reply packages/ez-react/src/docs/4_3_commonProps.stories.mdx:223]: done thanks!

File: packages/ez-react/src/docs/4_3_commonProps.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+  {
+    gdp: 4.872,
+    country: 'Japan',
+  },
+];`}
+  ></Tab>
+</Tabs>
+
+The column chart in this case is the best option.
+We should specify the domainKey of `xAxis` and `yAxis` to the corresponding name of the property.
```
[the remark this answers is not in the document]
Comment (nour-borgi) [reply packages/ez-react/src/docs/4_3_commonProps.stories.mdx:49]: Done! Thanks!

File: packages/ez-react/src/docs/4_3_commonProps.stories.mdx [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+In this section, we will cover properties that most of the charts have in common.
+These so-called props are responsible for the display and behavior of your chart.
+
+## Data
+
+When adding a chart to your web project,
+or creating a new chart recipe, We need to think about the fuel that we will use to fill up our engine, which is data.
+**EazyChart** handles it through its chart wrapper.
+
+So if you are creating some sort of visualization for your web project, all you need to do is to pass your data as a props to the chart.
```
[the remark this answers is not in the document]
Comment (nour-borgi) [reply packages/ez-react/src/docs/4_3_commonProps.stories.mdx:20]: done! thanks!

File: packages/ez-react/src/docs/4_2_responsiveCharts.stories.mdx [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+by default the chart comes with a `800x600` dimensions which can be overwritten if you wish to have a certain chart dimensions.
```
[the remark this answers is not in the document]
Comment (nour-borgi) [reply packages/ez-react/src/docs/4_2_responsiveCharts.stories.mdx:12]: done! thanks!

File: packages/ez-react/src/docs/4_2_responsiveCharts.stories.mdx (hunk 1 above)
Comment (nour-borgi) [reply packages/ez-react/src/docs/4_2_responsiveCharts.stories.mdx:11]: Thanks! done!

File: packages/ez-react/src/docs/4_3_commonProps.stories.mdx [hunk 8]
```diff
@@ -0,0 +1,226 @@
...
+
+> **NOTE**
+>
+> Providing the wrong property name or inaccurate values with inconsistent types to the axes may cause the chart to not be displayed correctly.
+
+## AnimationOptions
+
+The **EazyChart** library comes with a set of animations that you can use on your chart.
+
+As you may know, data analysis can be quite boring … let’s say animation makes it less boring.
```
[the remark this answers is not in the document]
Comment (nour-borgi) [reply packages/ez-react/src/docs/4_3_commonProps.stories.mdx:87]: done! thanks :)

File: packages/ez-react/src/docs/4_3_commonProps.stories.mdx [hunk 9]
```diff
@@ -0,0 +1,226 @@
...
+
+Column Chart without padding:
+(Column chart without padding)[link to the image]
+
+Column Chart with padding:
+(Column chart with padding)[link to the image]
+
+## Dimensions
+
+By default, Charts recipes in EazyChart have 800x600 pixels dimensions that can be overwritten.
```
[the remark this answers is not in the document]
Comment (nour-borgi) [reply packages/ez-react/src/docs/4_3_commonProps.stories.mdx:159]: done thanks!

**Review by marrouchi — APPROVED** [review]

**Pull request merged**